nixos/modules/services/networking/keepalived.nix

```diff
@@ -0,0 +1,53 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.keepalived;
+  keepalivedConf = pkgs.writeText "keepalived.conf" ''
+    ${cfg.extraConfig}
+  '';
+  pidFile = "/run/keepalived.pid";
+
+in
+{
+
+  options = {
+    services.keepalived = {
+
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to enable Keepalived.
+        '';
+      };
+
+      extraConfig = mkOption {
+        type = types.nullOr types.lines;
+        default = null;
```

Comment:
I think this default value will fail to evaluate (unless explicitly converted to a string with `toString`). I think just using an empty string here is okay anyway.